.travis.yml

```diff
@@ -0,0 +1,135 @@
+jobs:
+  include:
+      # Linux Jobs
+    - os: linux
+      language: python
+      python: 3.6
+      install:
+        - pip3 install --upgrade pip
+        - pip3 install -r ./.travis/travis-requirements.txt
+        - pip3 install pycairo
+        - pip3 install .
+      script:
+        - pytest
+
+    - os: linux
+      language: python
+      python: 3.7
+      install:
+        - pip3 install --upgrade pip
+        - pip3 install -r ./.travis/travis-requirements.txt
+        - pip3 install pycairo
+        - pip3 install .
+      script:
+        - pytest
+
+    - os: linux
+      language: python
+      python: 3.8
+      install:
+        - pip3 install --upgrade pip
+        - pip3 install -r ./.travis/travis-requirements.txt
+        - pip3 install pycairo
+        - pip3 install .
+      script:
+        - pytest
+
+      # MacOS (OSX) Jobs
+    - os: osx
+      language: sh
+      python: 3.6
+      env: PYVER="3.6.10"
+      install:
+        - pip3 install --upgrade pip
+        - pip3 install -r ./.travis/travis-requirements.txt
+        - pip3 install pycairo
+        - pip3 install .
+      script:
+        - pytest
+
+    - os: osx
+      language: sh
+      python: 3.7
+      env: PYVER="3.7.7"
+      install:
+        - pip3 install --upgrade pip
+        - pip3 install -r ./.travis/travis-requirements.txt
+        - pip3 install pycairo
+        - pip3 install .
+      script:
+        - pytest
+
+    - os: osx
+      language: sh
+      python: 3.8
+      env: PYVER="3.8.0"
```

Comment:
In previous iterations of this branch, I had to fix the 3.8 version to 3.8.0 because it would error out (I think the culprit was rich). We should document why we are doing this (since the latest version is 3.8.3), as well as periodically try again to bump this version.